pages/studentPortal/web/React/radix-ui.tsx

```diff
@@ -0,0 +1,128 @@
+import { Root, Indicator } from "@radix-ui/react-progress";
+import {
+  Provider,
+  Root as TooltipRoot,
+  Trigger,
+  Portal,
+  Content,
+  Arrow,
+} from "@radix-ui/react-tooltip";
+import { PlusCircleIcon, XIcon } from "@heroicons/react/outline";
+import {
+  Root as DialogRoot,
+  Trigger as DialogTrigger,
+  Portal as DialogPortal,
+  Overlay,
+  Content as DialogContent,
+  Title,
+  Description,
+  Close,
+} from "@radix-ui/react-dialog";
+import React, { useEffect, useState } from "react";
+import TooltipComponent from "../../../../components/coding/studentPortal/radix/TooltipComponent";
+import ProgressBarComponent from "../../../../components/coding/studentPortal/radix/ProgressBarComponent";
+import ProgressSlider from "../../../../components/coding/studentPortal/radix/ProgressSlider";
+
+const ProgressDemo = () => {
+  const [progress, setProgress] = useState(13);
+
+  useEffect(() => {
+    const timer = setTimeout(() => setProgress(66), 500);
+    return () => clearTimeout(timer);
+  }, []);
+
+  return (
+    <div className="space-y-10">
+      <ProgressBarComponent
+        elapsedDays={24}
+        totalTrialDays={30}
+        size={"medium"}
+      />
+      <ProgressSlider />
+      <div className="items-center bg-gray-400">
+        This is the progress bar
+        <Root
+          className="relative overflow-hidden bg-blue-400 rounded-full w-[300px] h-[25px]"
+          style={{
+            transform: "translateZ(0)",
+          }}
+          value={66}
+        >
+          <Indicator
+            className="bg-white w-full h-full transition-transform duration-[660ms] ease-[cubic-bezier(0.65, 0, 0.35, 1)]"
+            style={{ transform: `translateX(-${100 - progress}%)` }}
+          />
+        </Root>
+      </div>
+      <TooltipComponent message="Help me." active={false}>
+        <button className="text-blue-500 shadow-black-500 hover:bg-purple-200 inline-flex h-[35px] w-[35px] items-center justify-center rounded-full bg-white shadow-[0_2px_10px] outline-none focus:shadow-[0_0_0_2px] focus:shadow-black-300">
```

Comment:
These tailwind styles are a lot to take in. Can we extract this button into a resuable component and put it in a different file?

Reply:
This button I won't use in production... I want to wrap the tooltip component around the sidebar links.